src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -95,6 +100,69 @@ public static Email parseEmail(String email) throws ParseException {
         return new Email(trimmedEmail);
     }
 
+    /**
+     * Parses a {@code String eventDescription} into an {@code EventDescription}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code eventDescription} is invalid.
+     */
+    public static EventDescription parseEventDescription(String eventDescription) throws ParseException {
+        requireNonNull(eventDescription);
+        String trimmedEventDescription = eventDescription.trim();
+        if (!EventDescription.isValidEventDescription(trimmedEventDescription)) {
+            throw new ParseException(EventDescription.MESSAGE_CONSTRAINTS);
+        }
+        return new EventDescription(eventDescription);
+    }
+
+    /**
+     * Parses a {@code String date} into a {@code LocalDate}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code date} is invalid.
+     */
+    public static LocalDate parseDate(String date) throws ParseException {
+        requireNonNull(date);
+        String trimmedDate = date.trim();
+        try {
+            return LocalDate.parse(trimmedDate);
+        } catch (DateTimeParseException e) {
+            throw new ParseException(e.getMessage());
+        }
+    }
+
+    /**
+     * Parses a {@code String time} into an {@code LocalTime}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code time} is invalid.
+     */
+    public static LocalTime parseTime(String time) throws ParseException {
+        requireNonNull(time);
+        String trimmedTime = time.trim();
+        try {
+            return LocalTime.parse(trimmedTime);
+        } catch (DateTimeParseException e) {
+            throw new ParseException(e.getMessage());
+        }
+    }
+
+    /**
+     * Parses a {@code String duration} into an {@code Duration}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code duration} is invalid.
+     */
+    public static Duration parseDuration(String duration) throws ParseException {
+        requireNonNull(duration);
+        String trimmedDuration = duration.trim();
+        try {
+            return Duration.ofHours(Integer.parseInt(trimmedDuration));
```

Comment:
What if the duration is 90 minutes long?

Reply:
For now, it is set to only accept hours. I will be updating this later on.